src/main/java/bartenderbob/InputHandler.java

```diff
@@ -0,0 +1,168 @@
+ */
+public class InputHandler {
+    /**
+     * Handles the user input and returns the appropriate response.
+     * @param taskList The TaskList object that stores the tasks.
+     * @param ui The UI object that handles the user interface.
+     * @param userInput The user input.
+     * @return The response to the user input.
+     */
+    public static String handleInput(TaskList taskList, Ui ui, String userInput) {
+        assert taskList != null : "Task list cannot be null";
+        assert ui != null : "ui cannot be null";
+        assert userInput != null : "user input cannot be null";
+        String[] stringComponents = userInput.split(" ");
+        String firstWord = stringComponents[0];
+        if (firstWord.equals("bye")) {
+            return ui.leave();
+        } else {
+            try {
+                return processCommand(firstWord, stringComponents, userInput, taskList);
+            } catch (BartenderBobException e) {
+                //This exception is when the index from user input is out of the TASKS bounds.
+                return e.tasksOutOfBounds();
+            } catch (IndexOutOfBoundsException e) {
+                //IndexOutOfBoundsException is for missing userInputs.
+                BartenderBobException error = new BartenderBobException(firstWord);
+                return error.displayError();
+            } catch (IllegalArgumentException e) {
+                return ui.showInvalidDateFormat();
+            }
+        }
+    }
+
+    /**
+     * Processes the user input and returns the appropriate response.
+     * @param firstWord The first word of the user input.
+     * @param stringComponents The components of the user input.
+     * @param userInput The user input.
+     * @param taskList The TaskList object that stores the tasks.
+     * @return The response to the user input.
+     * @throws BartenderBobException If the user input is invalid.
+     */
+    public static String processCommand(String firstWord,
+                                        String[] stringComponents,
+                                        String userInput, TaskList taskList) throws BartenderBobException {
+        switch (firstWord) {
+        case "list":
+            return processListCommand(taskList);
+        case "mark":
+            return processMarkCommand(stringComponents, taskList);
+        case "unmark":
+            return processUnmarkCommand(stringComponents, taskList);
+        case "delete":
+            return processDeleteCommand(stringComponents, taskList);
+        case "find":
+            return processFindCommand(stringComponents, taskList);
+        case "todo":
+            return processTodoCommand(userInput, taskList);
+        case "deadline":
+            return processDeadlineCommand(userInput, taskList);
+        case "event":
+            return processEventCommand(userInput, taskList);
+        default:
+            return BartenderBobException.invalidInput(userInput);
```

Comment:
I like that you keep all the cases togethers 👍 and compact, increasing readability